docs/README.md

```diff
@@ -1530,13 +1537,13 @@ $ http --cert=client.pem --cert-key=client.key --cert-key-pass=my_password https
 ### SSL version
 
 Use the `--ssl=<PROTOCOL>` option to specify the desired protocol version to use.
-This will default to SSL v2.3 which will negotiate the highest protocol that both the server and your installation of OpenSSL support.
-The available protocols are `ssl2.3`, `ssl3`, `tls1`, `tls1.1`, `tls1.2`, `tls1.3`.
+This will default to TLS v1.0 which will negotiate the highest protocol that both the server and your installation of OpenSSL support.
+The available protocols are `tls1`, `tls1.1`, `tls1.2`, `tls1.3`.
```

Comment:
Love the idea, and without digging any deeper into the changeset to verify my thoughts in implementation, I'm just raising it here for general discussion:

"This is will default to TLS v1.0 which will negotiate the highest protocol that both […] support."

1. This sounds a bit confusing. It's either "This is will default to TLS v1.0" (which it should not), or "This is will default to a highest TLS version that both […]".
 - Because the values don't mean "use as the lowest", but "use the chosen only" (i.e. not a tls12+ or a range, but tls12 _only_)
 - I'd actually propose a magic value like "`tls`" (roughly meaning an alias to `PROTOCOL_TLS_CLIENT`) that would use the highest protocol available. Not sure if that's the default thou, not sure if the client can have tls13 capability, but still use tls12 as a default? If default == max always, then there's prolly no need for such value, only way to use max protocol then is to omit the param though. Which would mean no "max" magic value would be listed in help/man among the "available protocols", which in turn might seem a bit confusing then. So an explicit value for max protocol available, even when it really doesn't do anything if that's the default, can still have its value for clarity.
  - Do you plan to silently accept the old "`ssl2.3`" value mapped to default, whatever that ends up to be, not to raise any errors?

3. Are you sure the "`tls1.3`" value works? It does not in master, and never did (since the openssl apis are different for TLSv13, and the change in flags structure [see 4. ↓] made it impossible to constrain without all `*_OP_NO_*` used) — it only selects TLSv13 if that would have been the default anyways…

4. Adding ranges, i.e. min=tls11 max=tls12, or just one constraint alone (to match the current python sslcontext, instead of selecting one version, or disabling individual one by one) would also be lovely at some point.

Reply:
Indeed, the "This is will default to ..." is kind of confusing. I've addressed that by rephrasing it.

1) It is rephrased accordingly with "If not specified, it tries to negotiate the highest protocol that both the server and your installation of OpenSSL support."
2) "tls1.3" case has been addressed, we are now assuming that "tls1.3" (associated to PROTOCOL_TLS_CLIENT) means strictly do tls 1.3, and appropriate measures have been made.
3) The range of accepted tls version should be dealt with separately, I don't want to further increase this PR size, thus delaying it further on. I agree it can be useful, but it's not top priority as we speak.

 regards,

Reply:
To demonstrate `--ssl=tls1.3` is working, run `https --ssl=tls1.3 https://tls-v1-2.badssl.com:1012/`